### PR TITLE
feat(enterprise-baas-support): UI-31497: Scope in backup policy

### PR DESCRIPTION
**Pull request by sreekarbvibm**
Scope in create and get backup policy

SRB: https://github.ibm.com/cloudlab/srb/blob/dfcb509e0a3824321f93d00d70933ff63d7f2a97/proposals/3287/README.md
api-spec: https://github.ibm.com/riaas/api-spec/pull/4773/files
https://pages.github.ibm.com/vpc/vpc-spec-artifacts/branch/master/swagger-ui.html?feature=is-backup-policy-enterprise-baas#/Backup%20policies/create_backup_policy

### CODE REVIEW

File: ibm/service/vpc/resource_ibm_is_backup_policy.go [hunk 1]
```diff
@@ -88,6 +88,44 @@ func ResourceIBMIsBackupPolicy() *schema.Resource {
...
+			},
+			"scope": &schema.Schema{
+				Type:        schema.TypeList,
+				Computed:    true,
+				Optional:    true,
+				Description: "The scope for this backup policy.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"crn": &schema.Schema{
+							Type:        schema.TypeString,
```
[the remark this answers is not in the document]
Comment (sreekarbvibm) [reply ibm/service/vpc/resource_ibm_is_backup_policy.go:112]: Actually it shouldn't be optional - scope, health_state and health_reasons are all mandatory

**Review by deepaksibm — DISMISSED** [review]
add doc changes for backup policy resource

File: ibm/service/vpc/resource_ibm_is_backup_policy.go [hunk 2]
```diff
@@ -88,6 +88,47 @@ func ResourceIBMIsBackupPolicy() *schema.Resource {
...
+				MaxItems:    1,
+				Description: "The scope for this backup policy.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"crn": &schema.Schema{
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "The CRN for this enterprise.",
+						},
+						"id": &schema.Schema{
```
Comment (deepaksibm) [inline ibm/service/vpc/resource_ibm_is_backup_policy.go:117]: id and resource type are response fields only. And for response fields, it should be marked as computed only.

File: ibm/service/vpc/resource_ibm_is_backup_policy.go [hunk 3]
```diff
@@ -155,6 +207,14 @@ func resourceIBMIsBackupPolicyCreate(context context.Context, d *schema.Resource
 		}
 		createBackupPolicyOptions.SetResourceGroup(&resourceGroup)
 	}
+	if _, ok := d.GetOk("scope"); ok {
+		bkpPolicyScopePrototypeMap := d.Get("scope.0").(map[string]interface{})
+		bkpPolicyScopePrototype := vpcv1.BackupPolicyScopePrototype{}
+		if bkpPolicyScopePrototypeMap["crn"] != nil {
```
Comment (deepaksibm) [inline ibm/service/vpc/resource_ibm_is_backup_policy.go:213]: check for empty string as well

**Comment by sreekarbvibm** [comment]
> add doc changes for backup policy resource

Done

File: ibm/service/vpc/resource_ibm_is_backup_policy.go [hunk 4]
```diff
@@ -88,6 +88,45 @@ func ResourceIBMIsBackupPolicy() *schema.Resource {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
+			"health_reasons": &schema.Schema{
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "The reasons for the current health_state (if any).",
+				Elem: &schema.Schema{
```
Comment (deepaksibm) [inline ibm/service/vpc/resource_ibm_is_backup_policy.go:95]: health_reasons have code, more_info and message inside. It should be added in the schema.

File: website/docs/r/is_backup_policy.html.markdown [hunk 5]
```diff
@@ -55,6 +58,7 @@ In addition to all argument references listed, you can access the following attr
 - `lifecycle_state` - (String) The lifecycle state of the backup policy.
 - `resource_type` - (String) The resource type.
 - `version` - Version of the BackupPolicy.
+- `scope` - Scope of this backup policy
```
Comment (deepaksibm) [inline website/docs/r/is_backup_policy.html.markdown:61]: add scope details and nested elements

File: ibm/service/vpc/resource_ibm_is_backup_policy.go [hunk 6]
```diff
@@ -127,6 +182,17 @@ func ResourceIBMIsBackupPolicyValidator() *validate.ResourceValidator {
 			MaxValueLength:             128,
 		},
 	)
+	validateSchema = append(validateSchema,
```
Comment (deepaksibm) [inline ibm/service/vpc/resource_ibm_is_backup_policy.go:185]: We don't need a validator for TypeList schema argument scope. This can be removed.

**Comment by sreekarbvibm** [comment]
moved upstream

**Pull request closed without merging**